test/pages/register-a-beacon/application-complete.test.tsx

```diff
@@ -56,7 +58,7 @@ describe("ApplicationCompletePage", () => {
       );
       const result = await getServerSideProps(context);
 
-      expect(result.props.reference).toBeUndefined();
```

Comment:
`result.props.reference` being undefined was throwing hard-to-trace errors.  I replaced it with an empty string. 

In the medium term, the reference number logic should be moved to the API so I didn't spend too much time thinking about this.